### ISSUE TITLE
Inconsistent capitalisation

### ISSUE THREAD

**Issue by MaxNoe** [open-gamma-ray-astro/gamma-astro-data-formats#118]
Most `HDUCLAS*` and header keywords are in all-caps, but some are not, like `edisp_2d`.

We should make that consistent

**Comment by cdeil** [comment]
Do you mean here?
https://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html

Or do we have lower-case in other places?

**Comment by cdeil** [comment]
For the headings in the spec it doesn't matter, OK to change to all-caps if you like. If so, would you be willing to send a PR?

For the HDU index table, I chose lower-case because I hate typing all-caps or mixed-capitalisation things.

I think Gammapy is the only user of that index table. The plan is more or less that it will be superseded by something new very soon, so I propose to leave as-is. People have produced such index tables with lower-case in the past years and looking at
https://github.com/gammapy/gammapy/blob/e09a437b89ec56d6c979eb7da613ae7331726f0d/gammapy/data/hdu_index_table.py#L116
I think Gammapy doesn't support all-caps.

We could change to all-caps for consistency within the spec, and make the Gammapy code do capitalisation-independent string match, by calling `.upper`. @MaxNoe - Do you want to do that? Or just leave as-is and then develop new and better DL3 format in the coming months / years?

**Comment by MaxNoe** [comment]
We should be either consistently upper or lowercase or case insensitive. A mixture will just confuse people.

**Comment by cdeil** [comment]
The problem is that I chose lower-case in
https://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html
about 3 years ago, and then about 1 or 2 years ago we added
https://gamma-astro-data-formats.readthedocs.io/en/latest/general/hduclass.html
and there chose upper case.

I think if we want consistency, we will have to go upper-case, since already decades ago they chose
```
HDUCLAS1= 'EVENTS  '
```
https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/events/ogip_94_003/ogip_94_003.html

So here's what I plan to do (even though I personally hate typing all-caps):
- Make a PR to the spec changing these to all upper-case, in the case of the index table adding a note that lower-case was used previously and matching should thus be case-insensitive.
- Make a PR in Gammapy to change to upper-case, and use `.upper` to also support existing tables that use lower-case

Note that complete consistency within the spec to lower- or upper-case everywhere won't be achieved.  Of course, a proposal to make everything (keynames, column names and values) lowercase or uppercase can be made again, but I'm not even sure it's possible, given that FITS has certain limitations and quirks.  A discussion on this in the context of the SED spec was already done here: #65
